src/main/java/com/example/bookstore/controller/BookController.java

```diff
@@ -44,7 +45,8 @@ public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
     }
 
     @PutMapping("/{id}")
-    public BookDto updateBookById(@PathVariable Long id, @RequestBody BookRequestDto requestDto) {
+    public BookDto updateBookById(@PathVariable Long id,
```

Comment:
```suggestion
    public BookDto updateBookById(@PathVariable @Positive Long id,
```
Don't forget to add @Validated under the controller